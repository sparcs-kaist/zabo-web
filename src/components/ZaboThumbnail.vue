<template>
  <div class="zaboThumbnail" :class="`row${parseInt( row )}`">
    <div class="thumbnail" @hover="zaboThumbnailDetailShow()" :style="style">
      <p>
        {{ zaboDetail.content }}
      </p>
    </div>
    <div class="thumbnailDetail">
    </div>
  </div>
</template>

<script>
export default {
  props: {
    zaboDetail: {
      type: Object,
      default: () => {}
    },
    number: {
      type: Number,
      default: 0
    },
    row: {
      type: Number,
      default: 0
    }
  },
  data() {
    return {};
  },
  computed: {
    style() {
      let style;
      if (typeof this.zaboDetail.posters[0] !== "undefined") {
        style = `background-image: url(${this.zaboDetail.posters[0].image});`;
      }
      return style;
    }
  },
  methods: {
    zaboThumbnailDetailShow() {
      document.getElementsByClassName("thumbnailDetail")[
        this.number
      ].style.display = "block";
    }
  }
};
</script>

<style scoped>
.zaboThumbnail.row1 .thumbnail,
.zaboThumbnail.row7 .thumbnail {
  display: none;
}

.zaboThumbnail.row2 .thumbnail,
.zaboThumbnail.row6 .thumbnail {
  width: 178px;
  height: 277px;
  margin: 0 5px 0 6px;
  z-index: 1001;
}

.zaboThumbnail.row3 .thumbnail,
.zaboThumbnail.row5 .thumbnail {
  width: 226px;
  height: 394px;
  margin: 0 5px 0 6px;
  z-index: 1002;
}

.zaboThumbnail.row4 .thumbnail {
  width: 298px;
  height: 464px;
  margin: 0 5px 0 6px;
  z-index: 1003;
}

.zaboThumbnail.row2 {
  margin-bottom: -40px;
}

.zaboThumbnail.row3 {
  margin-bottom: -20px;
}

.zaboThumbnail.row5 {
  margin-top: -20px;
}

.zaboThumbnail.row6 {
  margin-top: -40px;
}

.thumbnail {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  position: relative;
  background-color: rgba(217, 217, 217, 1);
  background-size: cover;
  background-repeat: no-repeat;
  background-position: 50%;
}

.thumbnailDetail {
  display: none;
}
</style>
