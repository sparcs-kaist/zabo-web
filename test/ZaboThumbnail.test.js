import { shallow } from "vue-test-utils";
import ZaboThumbnail from "../src/components/ZaboThumbnail";

describe("ZaboThumbnail.test.js", () => {
  let cmp;

  beforeEach(() => {
    cmp = shallow(ZaboThumbnail, {
      // Be aware that props is overridden using `propsData`
      propsData: {
        zaboDetail: {
          content: "This is content",
          posters: ["http://poster"]
        }
      }
    });
    cmp.setData({});
  });

  it("Props match!", () => {
    expect(cmp.vm.zaboDetail).toEqual({
      content: "This is content",
      posters: ["http://poster"]
    });
  });

  it("has the expected html structure", () => {
    expect(cmp.element).toMatchSnapshot();
  });
});
