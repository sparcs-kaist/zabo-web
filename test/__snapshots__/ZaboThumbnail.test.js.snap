// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ZaboThumbnail.test.js has the expected html structure 1`] = `
<div
  class="zaboThumbnail row0"
>
  <div
    class="thumbnail"
    style="background-image: url(undefined);"
  >
    <p>
      
      This is content
    
    </p>
  </div>
   
  <div
    class="thumbnailDetail"
  />
</div>
`;
